ui: metrics load

vladlos opened the pull request:
Optimised metrics data loading when sharing URLs to metrics graphs with large timescale

Resolves: #44836

Release note(ui): none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/44962)
<!-- Reviewable:end -->

dhartunian commented on pkg/ui/src/views/devtools/containers/raftMessages/index.tsx:
```diff
@@ -81,7 +81,7 @@ export class RaftMessages extends React.Component<RaftMessagesProps> {
   }
 
-  componentWillMount() {
+  componentDidMount() {
```
I know these are good changes to make but if you want to address React upgrade warnings, put them in a separate PR.

elkmaster commented on pkg/ui/src/views/cluster/containers/timescale/index.tsx:
```diff
@@ -207,14 +206,22 @@ class TimeScaleDropdown extends React.Component<TimeScaleDropdownProps, {}> {
 
-    selected.key = "Custom";
-    this.props.setTimeScale(selected);
+  setDatesByQueryParams = (dates?: timewindow.TimeWindow) => {
```
Main idea was to leave url human readable or even user generated. Model of timewindow consists from 3 fields, including labeling. Pass the whole object will be messy. And passing only part of that will require very similar calculations.

dhartunian commented:
@elkmaster test doesn't currently pass for me, can you take a look?

vladlos committed:
ui: metrics load

Optimised metrics data loading when sharing URLs to metrics graphs with large timescale

Resolves: #44836

Release note(ui): none

dhartunian commented:
bors r+
